x-pack/filebeat/input/salesforce/config_auth.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package salesforce
+
+import "errors"
+
+type authConfig struct {
+	// See: https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_flows.htm&type=5
+	// for more information about OAuth2 flows.
+	OAuth2 *OAuth2 `config:"oauth2"`
+}
+
+type OAuth2 struct {
+	// See: https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_username_password_flow.htm&type=5
+	UserPasswordFlow *UserPasswordFlow `config:"user_password_flow"`
+	// See: https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_jwt_flow.htm&type=5
+	JWTBearerFlow *JWTBearerFlow `config:"jwt_bearer_flow"`
+}
+
+type UserPasswordFlow struct {
+	Enabled *bool `config:"enabled"`
+
+	ClientID     string `config:"client.id"`
+	ClientSecret string `config:"client.secret"`
+	Password     string `config:"password"`
+	TokenURL     string `config:"token_url"`
+	Username     string `config:"username"`
+}
+
+type JWTBearerFlow struct {
+	Enabled *bool `config:"enabled"`
```

Comment:
Does Enabled have to be a pointer because of the config struct tag? If not, we could avoid this nil check and simply check for its value.

Reply:
This is to keep it aligned with the other inputs in filebeat. All use a pointer to bool. So for consistency we favoured this.